src/tests/test_solo_repl_dev.cpp

```diff
@@ -97,19 +99,37 @@ struct Runner {
     }
 };
 
-struct rdev_req : boost::intrusive_ref_counter< rdev_req > {
+struct Waiter {
+    std::atomic< uint64_t > expected_comp{0};
+    std::atomic< uint64_t > actual_comp{0};
+    folly::Promise< folly::Unit > comp_promise;
+
+    Waiter(uint64_t num_op) : expected_comp{num_op} {}
+    Waiter() : Waiter{SISL_OPTIONS["num_io"].as< uint64_t >()} {}
+
+    folly::Future< folly::Unit > start(std::function< void(void) > f) {
+        f();
+        return comp_promise.getFuture();
+    }
+
+    void one_complete() {
+        if ((actual_comp.fetch_add(1) + 1) >= expected_comp.load()) { comp_promise.setValue(); }
+    }
+};
+
+struct test_repl_req : public repl_req_ctx {
     sisl::byte_array header;
     sisl::byte_array key;
     sisl::sg_list write_sgs;
     sisl::sg_list read_sgs;
     int64_t lsn;
```

Comment:
There seems to be some redundant data between this and the base; should it just be public?

Reply:
this is just in test code, but yes I should remove this lsn and use base class lsn.